File: Python3/프로그래머스/2/42747. H－Index/H－Index.py
def solution(citations):
    c = citations
    for h in range(len(c),-1,-1):
        more = 0
        for compare in c:
            if compare >= h:
                more +=1
        # h번 이상 인용된 논문이 h편 이상이라면
        if more >= h:
            return h

File: C++/프로그래머스/2/42747. H－Index/H－Index.cpp
#include <vector>
#include <algorithm>  

using namespace std;

int solution(vector<int> citations) {
    // 논문 인용 횟수를 내림차순으로 정렬
    sort(citations.begin(), citations.end(), greater<int>());

    int hIndex = 0;

    // 정렬된 인용 횟수를 이용해 H-Index 계산
    for (int i = 0; i < citations.size(); i++) {
        // 현재 논문이 i번째일 때, 인용 횟수가 i+1 이상이어야 함
        if (citations[i] >= i + 1) {
            hIndex = i + 1;  // H-Index 갱신
        } else {
            break;  // 더 이상 조건을 만족하지 않으면 종료
        }
    }

    return hIndex;
}
